### DISCUSSION TITLE
Relating the PointNet++ classification model to the original paper architecture description (part 2)

### DISCUSSION THREAD

**Discussion by DanielTakeshi** [pyg-team/pytorch_geometric#4496] [Q&A] [answered]
In a prior discussion https://github.com/pyg-team/pytorch_geometric/discussions/3924 I had asked about connection the PointNet++ architecture to what is in the original paper. The response there was very helpful, thank you very much!

In this next post, I just realized that there's something else I'm a little confused about. In the PointNet++ paper, it talks about MSG and MRG architectures and in the appendix, says:

![150645740-dfe33f55-a771-4028-8f9f-11534a932ea9](https://user-images.githubusercontent.com/5532096/163849252-f390d4eb-4e30-4c77-aa56-42e58031d896.png)

Does PyTorch Geometric use these variants? It seems like the classification variant ([implemented here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/pointnet2_classification.py)) does not use these and instead uses the first network architecture in the above screenshot. By going through the network step by step, it seems like it's entirely explained by the set abstraction modules. There is some "concatenation" in the sense that `x = self.conv(...)` will use both `x` and `pos` but I don't think that's what is intended from the MSG or MRG variants.

Is my understanding correct that if we want to try MSG and MRG variants, that we will need to adjust the code accordingly?

**Comment by rusty1s** [comment, marked as answer]
Yes, the example does not include MSG and MRG variants, sorry about that. I would definitely accept a PR that integrate these :)

**Reply by DanielTakeshi** [reply]
Thanks!
